src/guide-parallel.md

```diff
@@ -0,0 +1,144 @@
+    RNGs have been critiqued in that multiple streams using the same key are
+    often strongly correlated. See the [author's own comments](https://www.pcg-random.org/posts/critiquing-pcg-streams.html).
+
+    The PCG RNGs *also* support an `fn advance(delta)` method, which might be
+    used to divide a single stream into multiple sub-streams as with Xoshiro's
+    `jump` (but better since the offset can be specified).
+
+## Practice: non-deterministic multi-threaded
+
+We use [Rayon]'s parallel iterators, using [`map_init`] to initialize an RNG in
+each worker thread. Note: this RNG may be re-used across multiple work units,
+which may be split between worker threads in non-deterministic fashion.
+
+```rust
+use rand::distributions::{Distribution, Uniform};
+use rayon::prelude::*;
+
+static SAMPLES: u64 = 1_000_000;
+
+fn main() {
+    let range = Uniform::new(-1.0f64, 1.0);
+
+    let in_circle = (0..SAMPLES)
+        .into_par_iter()
+        .map_init(|| rand::thread_rng(), |rng, _| {
+            let a = range.sample(rng);
+            let b = range.sample(rng);
+            if a * a + b * b <= 1.0 {
+                1
+            } else {
+                0
+            }
+        })
+        .reduce(|| 0usize, |a, b| a + b);
+
+    // prints something close to 3.14159...
+    println!(
+        "π is approximately {}",
+        4. * (in_circle as f64) / (SAMPLES as f64)
+    );
+}
+```
+
+## Practice: determinsitic multi-threaded
+
+We use approach (4) above to achieve a deterministic result: initialize all RNGs
+from a single seed, but using multiple streams.
+We use [`ChaCha8Rng::set_stream`] to achieve this.
+
+Note further that we manually batch multiple work-units according to
+`BATCH_SIZE`. This is important since the cost of initializing an RNG is large
+compared to the cost of our work unit (generating two random samples plus some
+trivial calculations). Manual batching could improve performance of the above
+non-deterministic simulation too.
+
+(Note: this example is <https://github.com/rust-random/rand/blob/master/examples/rayon-monte-carlo.rs>.)
+
+```rust
+use rand::distributions::{Distribution, Uniform};
+use rand_chacha::{rand_core::SeedableRng, ChaCha8Rng};
+use rayon::prelude::*;
+
+static SEED: u64 = 0;
+static BATCH_SIZE: u64 = 10_000;
+static BATCHES: u64 = 1000;
+
+fn main() {
+    let range = Uniform::new(-1.0f64, 1.0);
+
+    let in_circle = (0..BATCHES)
+        .into_par_iter()
+        .map(|i| {
+            let mut rng = ChaCha8Rng::seed_from_u64(SEED);
+            rng.set_stream(i);
+            let mut count = 0;
+            for _ in 0..BATCH_SIZE {
+                let a = range.sample(&mut rng);
+                let b = range.sample(&mut rng);
+                if a * a + b * b <= 1.0 {
+                    count += 1;
+                }
+            }
+            count
+        })
+        .reduce(|| 0usize, |a, b| a + b);
+
+    // prints something close to 3.14159...
```

Comment:
You could assert an exact value here, to highlight the determinism.

Reply:
Good idea. For reference, the value for this seed is 3.1409052